apps/mark-scan/fai-100-controller/src/fai_100_controllerd.rs

```diff
@@ -328,6 +328,30 @@ fn handle_status_response(
     let new_puff_status = new_status.puff_status;
     let new_connection_status = new_status.sip_puff_device_connection_status;
 
+    // When no sip & puff device is connected, the sip/puff values contradict docs.
+    // +-------------------------+-------------------+-------------+
+    // | Device Status           | Sip Status        | Value (Hex) |
+    // +-------------------------+-------------------+-------------+
+    // | Device Connected        | No Sip            | 0x00        |
+    // | Device Connected        | Sip               | 0x01        |
+    // | Device Not Connected    | No Sip            | 0x01        |
+    // | Device Not Connected    | Sip               | 0x00        |
+    // +-------------------------+-------------------+-------------+
+    // To avoid processing erroneous signals, we throw out sip/puff data for a short time
+    // after device connection status changes
+    if new_connection_status != current_status.sip_puff_device_connected {
+        current_status.sip_puff_device_connected = new_connection_status;
+        current_status.sip = SipAndPuffSignalStatus::Idle;
+        current_status.puff = SipAndPuffSignalStatus::Idle;
+        current_status.button_pressed = ButtonSignal::NoButton;
+
+        // Write device connection status to system file so mark-scan app is aware
+        write_pat_connection_status(new_connection_status, workspace_path)?;
+
+        sleep(2 * POLL_INTERVAL);
```

Comment:
I've tested this on hardware and it seems to work well.

The buffer doesn't queue anything unless we send the `GetNotificationStatus` command. That command itself is a misnomer, because there are no notifications or events. Rather, the device responds with a snapshot of the currently activated inputs and PAT connection status. It sends this snapshot exactly once in response to `GetNotificationStatus`, so status reports should map exactly 1:1 with `GetNotificationStatus` commands sent and we shouldn't have to worry about enqueued status reports.

When we detect PAT connection status at has changed we do 2 things

1. Wait for a period so `sip status` and `puff status` can settle
2. Throw out the signals from the previous response because we know we can't trust it

I think to make this more readable and clear, we can change `handle_status_response` to return a `bool` describing whether the next poll should be delayed. We can check that and sleep right before the next `GetNotificationStatus` command is sent. This locates the `sleep` right next to the thing we actually want to delay.